### PR TITLE
Add delayed and waiting state to paasta status for marathon jobs

### PR DESCRIPTION
**Pull request by keshavdv**
screen shot 2016-06-21 at 4 24 09 pm
screen shot 2016-06-21 at 4 23 56 pm

### CODE REVIEW

File: paasta_tools/marathon_serviceinit.py [hunk 1]
```diff
@@ -103,13 +103,29 @@ def status_desired_state(service, instance, client, job_config):
...
+
+
 def status_marathon_job(service, instance, app_id, normal_instance_count, client):
     name = PaastaColors.cyan(compose_job_id(service, instance))
     if marathon_tools.is_app_id_running(app_id, client):
         app = client.get_app(app_id)
         running_instances = app.tasks_running
+        is_overdue, backoff_seconds = get_app_queue_status(app_id, client)
+
         if len(app.deployments) == 0:
```
Comment (solarkennedy) [inline paasta_tools/marathon_serviceinit.py]: This now is too early, right? The order here will mater, this probably needs to go further down in the if clause

File: paasta_tools/marathon_serviceinit.py [hunk 2]
```diff
@@ -103,13 +103,29 @@ def status_desired_state(service, instance, client, job_config):
...
             deploy_status = PaastaColors.bold("Running")
+        elif is_overdue:
+            deploy_status = PaastaColors.red("Waiting for resources")
+        elif backoff_seconds:
+            delay_time = PaastaColors.grey("%s seconds" % backoff_seconds)
+            deploy_status = "%s (%s)" % (PaastaColors.red("Delayed"), delay_time)
         else:
```
Comment (solarkennedy) [inline paasta_tools/marathon_serviceinit.py]: Can you add one more?

```
elif app.instances == 0:
  deploy_status = PaastaColors.grey("Stopped")
```

File: paasta_tools/marathon_serviceinit.py [hunk 3]
```diff
@@ -103,13 +103,29 @@ def status_desired_state(service, instance, client, job_config):
     return "State:      %s - Desired state: %s" % (status, desired_state)
 
 
+def get_app_queue_status(app_id, client):
```
Comment (Rob-Johnson) [inline paasta_tools/marathon_serviceinit.py]: could you move this into marathon_tools please? also, please swap the order of the params to make it ``client, app_id``

**Comment by Rob-Johnson** [comment]
shipit

File: paasta_tools/marathon_serviceinit.py [hunk 4]
```diff
@@ -108,10 +108,23 @@ def status_marathon_job(service, instance, app_id, normal_instance_count, client
...
         else:
-            deploy_status = PaastaColors.yellow("Deploying")
+            # App is currently deploying so we should check the launch queue for more info
+            is_overdue, backoff_seconds = marathon_tools.get_app_queue_status(client, app_id)
+
+            if is_overdue:
+                deploy_status = PaastaColors.red("Waiting for resources")
+            elif backoff_seconds:
+                delay_time = PaastaColors.grey("%s seconds" % backoff_seconds)
+                deploy_status = "%s (%s)" % (PaastaColors.red("Delayed"), delay_time)
```
Comment (solarkennedy) [inline paasta_tools/marathon_serviceinit.py]: I would like to make this more obvious to users what Waiting and Delayed means, but I can't think of good signals.

Any ideas on making this more verbose?

Maybe:

   "Waiting: new tasks are not launching due to lack of capacity"

and

   "Delayed: Next task won't launch for %s seconds due to previous failures"

?

**Comment by solarkennedy** [comment]
shipit

**Comment by solarkennedy** [comment]
shipit

**Pull request merged**